caikit/interfaces/runtime/service.py

```diff
@@ -0,0 +1,242 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains interfaces and descriptor functions for service generation
+"""
+
+# Standard
+from enum import Enum
+from typing import Optional, Type, Union
+
+# First Party
+import alog
+
+# Local
+from caikit.config import get_config
```

Comment:
Same comment about relative imports (no change needed necessarily, just noting it)